File: app/src/main/java/cn/hukecn/fund/MyHttp.java
package cn.hukecn.fund;

import android.content.Context;

import com.loopj.android.http.AsyncHttpClient;
import com.loopj.android.http.AsyncHttpResponseHandler;

import org.apache.http.Header;

import java.io.UnsupportedEncodingException;

/**
 * Created by Kelson on 2016/1/6.
 */
public class MyHttp {
    static AsyncHttpClient client = new AsyncHttpClient();

    public static void get(Context context, String url, final AsyncHttp.HttpListener listener, final String charset){
        client.setTimeout(10000);
        client.setThreadPool(AppThreadPool.getInstance().getPool());
        client.setResponseTimeout(10000);
        client.get(context, url, new AsyncHttpResponseHandler() {
            @Override
            public void onSuccess(int i, Header[] headers, byte[] bytes) {
                try {
                    if(listener != null && bytes != null)
                        listener.onHttpCallBack(i,new String(bytes,charset));
                } catch (UnsupportedEncodingException e) {}
            }

            @Override
            public void onFailure(int i, Header[] headers, byte[] bytes, Throwable throwable) {
                //
            }
        });
    }
}

File: app/src/main/java/cn/hukecn/fund/MyHttp.kt
package cn.hukecn.fund

import android.content.Context
import cn.hukecn.fund.AsyncHttp.HttpListener
import com.loopj.android.http.AsyncHttpClient
import com.loopj.android.http.AsyncHttpResponseHandler
import org.apache.http.Header
import java.io.UnsupportedEncodingException
import java.nio.charset.Charset

/**
 * Created by Kelson on 2016/1/6.
 */
object MyHttp {
    var client = AsyncHttpClient()
    operator fun get(context: Context?, url: String?, listener: HttpListener?, charset: String?) {
        client.setTimeout(10000)
        client.threadPool = AppThreadPool.instance!!.pool
        client.responseTimeout = 10000
        client[context, url, object : AsyncHttpResponseHandler() {
            override fun onSuccess(i: Int, headers: Array<Header>, bytes: ByteArray) {
                try {
                    if (listener != null && bytes != null) listener.onHttpCallBack(i, String(bytes, Charset.forName(charset)))
                } catch (e: UnsupportedEncodingException) {
                }
            }

            override fun onFailure(i: Int, headers: Array<Header>, bytes: ByteArray, throwable: Throwable) {
                //
            }
        }]
    }
}